Framework/API/src/Algorithm.cpp

```diff
@@ -1524,7 +1517,6 @@ void Algorithm::cancel() {
  * and check if the algorithm has requested that it be cancelled.
  */
 void Algorithm::interruption_point() {
-  Poco::FastMutex::ScopedLock _lock(m_mutex);
```

Comment:
This certainly does seem completely unnecessary given the `IF_NOT_PARALLEL` below. I'm not sure why that is required either but I think that is a question for another day.